agent_build/tools/environment_deployments/deployments.py

```diff
@@ -610,11 +617,193 @@ def get_deployment_by_name(name: str) -> Deployment:
+            shutil.rmtree(step_output_dir_path)
+
+        hit = False
+        if common.IN_CICD:
+            hit = self._restore_from_cache("output_registry", step_output_dir_path)
+            if hit:
+                logging.info(
+                    "The local registry with base docker image is found in cache. Reuse it instead of building."
+                )
+                return
+
+        logging.info("Build docker base image.")
+        if step_output_dir_path.is_dir():
+            shutil.rmtree(step_output_dir_path)
+        step_output_dir_path.mkdir(parents=True)
+
+        def _build_base_image(
+            result_image_name: str,
+            with_coverage: bool
+        ):
+            dockerfile_path = self._isolated_source_root_path / _REL_AGENT_BUILD_DOCKER_PATH / "Dockerfile.base"
+
+            command_args = [
+                "docker",
+                "buildx",
+                "build",
+                "-t",
+                f"localhost:5000/{result_image_name}",
+                "-f",
+                str(dockerfile_path),
+                "--push",
+                "--build-arg",
+                f"BASE_IMAGE_SUFFIX={type(self).BASE_IMAGE_NAME_SUFFIX}"
+            ]
+
+            if with_coverage:
+                command_args.append("--build-arg")
+                command_args.append("COVERAGE_VERSION=4.5.4")
+
+            # Add all needed platforms to build.
+            for plat in constants.AGENT_DOCKER_IMAGE_SUPPORTED_PLATFORMS:
+                command_args.append("--platform")
+                command_args.append(plat.value)
+
+            # Add final argument of the command - build context path.
+            # Specify here our special root directory which contains only
+            # tracked files that are tracked by that step.
+            command_args.append(
+                str(self._isolated_source_root_path)
+            )
+
+            # Run build.
+            common.check_call_with_log(
+                command_args
+            )
+
+        # Create registry where the result base image is pushed.
+        registry_container = build_in_docker.LocalRegistryContainer(
+            name="output_registry_path",
+            registry_port=5000,
+            # specify directory where registry's data root is mapped.
+            registry_data_path=step_output_dir_path
+        )
+
+        # Start registry and build and push base images within while it's running.
+        with registry_container:
+            # build testing version of the base image. (with coverage)
+            _build_base_image(
+                result_image_name=f"{type(self).BASE_IMAGE_NAME}_testing",
+                with_coverage=True
+            )
+            # build prod version of the base image.
+            _build_base_image(
+                result_image_name=type(self).BASE_IMAGE_NAME,
+                with_coverage=False
+            )
+
+        if common.IN_CICD and not hit:
```

Comment:
Does caching still works if you run build package script locally?

Reply:
Locally it is handled by docker's own caching. You just have to build it once, and docker will reuse cached layers, so there's no need to enable the `IN_CICD ` flag locally.